### PR TITLE
Fixed weapon specials text not displaying in correct color: Issue #4071

### PR DESCRIPTION
**Pull request by Leonard03**

### CODE REVIEW

File: src/reports.cpp [hunk 1]
```diff
@@ -934,7 +934,10 @@ static int attack_info(reports::context & rc, const attack_type &at, config &res
 	}
 
 	{
-		auto ctx = at.specials_context_for_listing(u.side() == rc.screen().playing_side());
+		//If we have a second unit, do the 2-unit specials_context
+		auto ctx = (sec_u == nullptr) ? at.specials_context_for_listing(u.side() == rc.screen().playing_side()) :
+						at.specials_context(unit_const_ptr(&u), unit_const_ptr(sec_u), hex, sec_u->get_location(), u.side() == rc.screen().playing_side(), sec_u_weapon);
```
Comment (jyrkive) [inline src/reports.cpp]: The check for "attacking" is duplicated across these two lines. Could you extract it into a named variable?

Comment (Leonard03) [reply src/reports.cpp]: Done.

**Comment by jyrkive** [comment]
Merged. Thank you! :)

**Comment by jostephd** [comment]
Thanks guys! Any chance of a backport?

**Comment by jyrkive** [comment]
This is an enhancement rather than a bugfix, which is why I decided not to backport it.

**Comment by jostephd** [comment]
For the record, I proposed the backport because I viewed this as a bugfix rather than an enhancement. :)

Anyway, I guess we'll treat it as an enhancement, so I added this patch to my custom patchset. I had to cherry-pick 826a76a939149409d5e98d9fc36664a92efc6438 first.

Thanks again for merging this.

**Pull request merged**